velox/functions/prestosql/tests/CastBaseTest.h

```diff
@@ -67,25 +71,37 @@ class CastBaseTest : public FunctionBaseTest {
     assertEqualVectors(expected, result);
   }
 
+  core::TypedExprPtr buildCastExprDictEncoding(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      bool tryCast = false) {
+    core::TypedExprPtr inputField =
+        std::make_shared<const core::FieldAccessTypedExpr>(fromType, "c0");
+    core::TypedExprPtr callExpr = std::make_shared<const core::CallTypedExpr>(
+        fromType,
+        std::vector<core::TypedExprPtr>{inputField},
+        "testing_dictionary");
+    return std::make_shared<const core::CastTypedExpr>(
+        toType, std::vector<core::TypedExprPtr>{callExpr}, tryCast);
+  }
+
+  void verifyCastDictEncoding(
+      const VectorPtr& expected,
+      const VectorPtr& actual) {
+    auto indices =
+        ::facebook::velox::test::makeIndicesInReverse(expected->size(), pool());
+    assertEqualVectors(
+        wrapInDictionary(indices, expected->size(), expected), actual);
```

Comment:
Consider adding helper function wrapInDictionary that takes only indices and base vector and uses indices->size() / sizeof(vector_size_t) for size.